src/Form/LinkBlockFormDataProvider.php

```diff
@@ -0,0 +1,298 @@
+<?php
+/**
+ * 2007-2018 PrestaShop.
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Academic Free License 3.0 (AFL-3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/AFL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/AFL-3.0 Academic Free License 3.0 (AFL-3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\Module\LinkList\Form;
+
+use PrestaShop\Module\LinkList\Cache\LinkBlockCacheInterface;
+use PrestaShop\Module\LinkList\Model\LinkBlock;
+use PrestaShop\Module\LinkList\Repository\LinkBlockRepository;
+use PrestaShop\PrestaShop\Core\Addon\Module\ModuleRepository;
+use PrestaShop\PrestaShop\Core\Form\FormDataProviderInterface;
+use Hook;
+use Ps_Linklist;
+
+/**
+ * Class LinkBlockFormDataProvider.
+ */
+class LinkBlockFormDataProvider implements FormDataProviderInterface
```

Comment:
"By the book" the validation should be done in [FormHandlerInterface](https://github.com/PrestaShop/PrestaShop/blob/develop/src/Core/Form/FormHandlerInterface.php#L41), this class have too many responsibilities.

I don't say you have to update it :+1: